### PR TITLE
Validate OpenAPI schema on docs job

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

**Comment by fao89** [comment]
**Requires:** https://github.com/pulp/pulpcore/pull/1065

File: templates/github/.github/workflows/scripts/script.sh.j2 [hunk 1]
```diff
@@ -26,6 +26,10 @@ if [[ "$TEST" = "docs" || "$TEST" = "publish" ]]; then
   make PULP_URL="http://pulp" html
   cd ..
 
+  # Validate OpenAPI schema
+  echo "Validating OpenAPI schema..."
+  cmd_prefix bash -c "django-admin spectacular --file pulp_schema.yml --validate"
```
Comment (mdellweg) [inline templates/github/.github/workflows/scripts/script.sh.j2:31]: Can this be `pulpcore-manager ...` ?

Comment (fao89) [reply templates/github/.github/workflows/scripts/script.sh.j2]: I tested locally and it works, I'm switching to `pulpcore-manager`

**Review by mdellweg — APPROVED** [review]

File: templates/github/.ci/scripts/schema.py.j2 [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"""
+Customizing OpenAPI validation.
+
+OpenAPI requires paths to start with slashes:
+https://spec.openapis.org/oas/v3.0.3#patterned-fields
+
+But some pulp paths starts with curly brackets e.g. {artifact_href}
+This script modifies drf-spectacular schema validation to accept slashes and curly brackets.
+"""
```
Comment (daviddavis) [inline templates/github/.ci/scripts/schema.py.j2:9]: :+1:

File: templates/github/.ci/scripts/schema.py.j2 [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Customizing OpenAPI validation.
+
+OpenAPI requires paths to start with slashes:
+https://spec.openapis.org/oas/v3.0.3#patterned-fields
+
+But some pulp paths starts with curly brackets e.g. {artifact_href}
```
Comment (mdellweg) [inline templates/github/.ci/scripts/schema.py.j2:7]:
```suggestion
But some pulp paths start with curly brackets e.g. "{artifact_href}"
```

File: templates/github/.ci/scripts/schema.py.j2 (hunk 2 above)
Comment (mdellweg) [reply templates/github/.ci/scripts/schema.py.j2:9]: I do not see any templating in this file. Maybe ditch the ".j2" or include the dont-touch-this header.

**Review by mdellweg — APPROVED** [review]
Just nit-picks

**Review by daviddavis — APPROVED** [review]

**Pull request merged**